### PR TITLE
[CI] Set up `compatibility test` workflow

### PR DESCRIPTION
**Pull request by rmanaem**
## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [ ] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [x] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: .github/workflows/compatibility-test.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Install cypress
+        run: npm install
+
+      - name: wait for graph set up
+        working-directory: ./dev
+        run: |
+          while ! tail -n 0 -f ./DEPLOY.log | grep -q "Finished setting up the Neurobagel graph backend."; do
+            :
+          done
+          echo "Finished setting up the Neurobagel graph backend."
```
Comment (surchs) [inline .github/workflows/compatibility-test.yaml:40]: :cherries: add empty line here?

File: cypress/fixtures/example.json
Comment (surchs) [inline cypress/fixtures/example.json:1]: don't need this?

**Review by surchs — APPROVED** [review]
looks very nice!

:cook:

File: cypress/fixtures/example.json
Comment (rmanaem) [reply cypress/fixtures/example.json:1]: No, removing

**Pull request merged**